tests/IResearch/IResearchViewCoordinatorTest.cpp

```diff
@@ -122,8 +120,7 @@ TEST_F(IResearchViewCoordinatorTest, test_rename) {
       "{ \"name\": \"testView\", \"type\": \"arangosearch\", \"id\": \"1\", "
       "\"collections\": [1,2,3] }");
 
-  Vocbase vocbase(TRI_vocbase_type_e::TRI_VOCBASE_TYPE_COORDINATOR,
-                  testDBInfo(server.server()));
+  Vocbase vocbase(testDBInfo(server.server()));
```

Comment:
This is a coordinator test. Is it clear that the code executed with this Vocbase object will behave the same as before? Does the code "feel" that it is supposed to run in coordinator mode here?

Reply:
Previously there have been some `if...else...` branches inside `TRI_vocbase_t` based on the vocbase's `_type` attribute. 
That means for vocbases of type `TRI_VOCBASE_TYPE_COORDINATOR` a few other branches were taken than for vocbases of type `TRI_VOCBASE_TYPE_NORMAL`. This now still happens, but it is not tied to the `_type` attribute anymore, but to the return value of `ServerState::instance()->isCoordinator()`.
There was a direct relationship between 
* `ServerState::instance()->isCoordinator()` => `TRI_VOCBASE_TYPE_COORDINATOR`
* `! ServerState::instance()->isCoordinator()` => `TRI_VOCBASE_TYPE_NORMAL`
As the unit test still passes with the changes from this PR, it probably does not make a difference.

At some point we should also get rid of all the `ServerState::instance()->isXXX()` throughout the code, because that IMO is also not ideal. But the `_type` attribute was yet redundant to the `ServerState` values, so it is for sure not needed.

Reply:
OK